### PR TITLE
new translation: forms/other_form_controls

### PR DESCRIPTION
**Pull request by jasonren0403**

### CODE REVIEW

File: files/zh-cn/learn/forms/other_form_controls/index.md [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+- [原生表单部件](/zh-CN/docs/Learn/Forms/Basic_native_form_controls)
+- [HTML5 input 类型](/zh-CN/docs/Learn/Forms/HTML5_input_types)
+- [其它表单控件](/zh-CN/docs/Learn/Forms/Other_form_controls)
+- [样式化 HTML 表单](/zh-CN/docs/Learn/Forms/Styling_web_forms)
+- [高级表单样式](/zh-CN/docs/Learn/Forms/Advanced_form_styling)
+- [UI 伪类](/zh-CN/docs/Learn/Forms/UI_pseudo-classes)
+- [客户端表单验证](/zh-CN/docs/Learn/Forms/Form_validation)
+- [发送表单数据](/zh-CN/docs/Learn/Forms/Sending_and_retrieving_form_data)
+
+### 高级技巧
```
Comment (jasonren0403) [inline files/zh-cn/learn/forms/other_form_controls/index.md:321]: 跟其他地方的翻译统一一下：
```suggestion
### 进阶内容
```

File: files/zh-cn/learn/forms/other_form_controls/index.md [hunk 2]
```diff
@@ -0,0 +1,325 @@
+---
+title: 其他表单部件
+slug: Learn/Forms/Other_form_controls
+---
+{{LearnSidebar}}{{PreviousMenuNext("Learn/Forms/HTML5_input_types","Learn/Forms/Styling_web_forms", "Learn/Forms")}}
+
+现在我们来具体看看非 `<input>` 表单部件的功能，从下拉列表菜单（drop-down list）、多行文本域（multi-line text field）到其他有用的表单功能，如我们在前一篇文章看到的 {{htmlelement('output')}} 元素和进度条（progress bar）元素。
```
Comment (yin1999) [inline files/zh-cn/learn/forms/other_form_controls/index.md:7]: We'd better use `表单控件` instead of `表单部件`

**Review by yin1999 — CHANGES_REQUESTED** [review]

File: files/zh-cn/learn/forms/other_form_controls/index.md [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+> **备注：** 你可以在该系列的第一篇文章中整理的例子中找到一个稍微有趣的文本区域使用的[示例](https://mdn.github.io/learning-area/html/forms/your-first-HTML-form/first-form-styled.html)（[顺便看看源代码](https://github.com/mdn/learning-area/blob/main/html/forms/your-first-HTML-form/first-form-styled.html)）。
+
+### 控制多行渲染
+
+{{htmlelement("textarea")}} 接受三种属性来控制其多行渲染行为：
+
+- {{htmlattrxref("cols","textarea")}}
+  - : 指定文本部件的可见宽度（列），单位为字符的宽度。这实际上是起始宽度，因为它可以通过调整 `<textarea>` 的大小来改变，也可以用CSS重写。如果没有指定，默认值是20。
+- {{htmlattrxref("rows","textarea")}}
+  - : 指定文本部件的可见行数。这实际上是起始高度，因为它可以通过调整 `<textarea>` 的大小来改变，也可以用CSS重写。如果没有指定，默认值是2。
```
Comment (yin1999) [inline files/zh-cn/learn/forms/other_form_controls/index.md:62]:
```suggestion
  - : 指定文本控件的可见宽度（列），单位为字符的宽度。这实际上是起始宽度，因为它可以通过调整 `<textarea>` 的大小来改变，也可以用 CSS 重写。如果没有指定，默认值是 20。
- {{htmlattrxref("rows","textarea")}}
  - : 指定文本控件的可见行数。这实际上是起始高度，因为它可以通过调整 `<textarea>` 的大小来改变，也可以用 CSS 重写。如果没有指定，默认值是 2。
```

File: files/zh-cn/learn/forms/other_form_controls/index.md [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+
+- `both`：默认值——允许横向和纵向缩放行为。
+- `horizontal`：只允许横向水平缩放行为。
+- `vertical`：只允许纵向竖直缩放行为。
+- `none`：不允许缩放行为。
+- `block` 和 `inline`：实验性的值，仅允许以 `block` 或 `inline` 方向缩放，如何变化取决于文字的方向，请参阅[处理不同文字方向](/zh-CN/docs/Learn/CSS/Building_blocks/Handling_different_text_directions)以了解更多内容。
+
+多多调试 {{cssxref("resize")}} 参考页面顶部的例子，来看看它们是如何运行的。
+
+## 下拉部件
```
Comment (yin1999) [inline files/zh-cn/learn/forms/other_form_controls/index.md:78]:
```suggestion
## 下拉控件
```

File: files/zh-cn/learn/forms/other_form_controls/index.md [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+
+下拉菜单是一种简单地让用户从许多不同选项中进行选择的部件，不占用用户界面太多空间。HTML 有两种下拉内容：一种是**选择框**、另外一种是**自动补全框**。这两种情况下的交互是相同的——一旦被激活，浏览器就会显示可供用户选择的值的列表。
```
Comment (yin1999) [inline files/zh-cn/learn/forms/other_form_controls/index.md:80]:
```suggestion
下拉菜单是一种简单地让用户从许多不同选项中进行选择的控件，不占用用户界面太多空间。HTML 有两种下拉内容：一种是**选择框**、另外一种是**自动补全框**。这两种情况下的交互是相同的——一旦被激活，浏览器就会显示可供用户选择的值的列表。
```

File: files/zh-cn/learn/forms/other_form_controls/index.md [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+     <option>Carrot</option>
+     <option>Eggplant</option>
+     <option>Potato</option>
+   </optgroup>
+</select>
+```
+
+{{EmbedLiveSample("多选选择框", 120, 120)}}
+
+> **备注：** 在多选选择框的情况下，你会注意到选择框不再以下拉内容的形式显示数值——相反，所有的值都会一次性显示在一个列表中，可选的 [`size`](/zh-CN/docs/Web/HTML/Attributes/size) 属性决定了部件的高度。
```
Comment (yin1999) [inline files/zh-cn/learn/forms/other_form_controls/index.md:162]:
```suggestion
> **备注：** 在多选选择框的情况下，你会注意到选择框不再以下拉内容的形式显示数值——相反，所有的值都会一次性显示在一个列表中，可选的 [`size`](/zh-CN/docs/Web/HTML/Attributes/size) 属性决定了控件的高度。
```

File: files/zh-cn/learn/forms/other_form_controls/index.md [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+
+> **备注：** 任何支持 {{HTMLElement("select")}} 元素的浏览器也支持 {{htmlattrxref("multiple","select")}} 属性。
+
+### 自动补充框
```
Comment (yin1999) [inline files/zh-cn/learn/forms/other_form_controls/index.md:166]:
```suggestion
### 自动补全框
```

File: files/zh-cn/learn/forms/other_form_controls/index.md [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+
+你可以通过 {{HTMLElement("datalist")}} 元素中的一些显示的 {{HTMLElement("option")}} 子元素为表单项提供推荐的自动完成值。`<datalist>` 元素需要指定一个 `id`。
```
Comment (yin1999) [inline files/zh-cn/learn/forms/other_form_controls/index.md:168]:
```suggestion
你可以通过 {{HTMLElement("datalist")}} 元素中的一些显示的 {{HTMLElement("option")}} 子元素为表单项提供推荐的自动补全值。`<datalist>` 元素需要指定一个 `id`。
```

File: files/zh-cn/learn/forms/other_form_controls/index.md [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+
+数据列表会使用 {{htmlattrxref("list","input")}} 属性绑定至一个 {{htmlelement("input")}} 元素（如 `text` 或 `email` 输入类型），该属性的取值就是要绑定的数据列表的 `id` 值。
+
+一旦数据列表与表单控件相关联，它的选项就会被用来自动完成用户输入的文本；通常，它以下拉框的形式呈现给用户，列出他们输入内容的可能匹配。
```
Comment (yin1999) [inline files/zh-cn/learn/forms/other_form_controls/index.md:172]:
```suggestion
一旦数据列表与表单控件相关联，它的选项就会被用来自动补全用户输入的文本；通常，它以下拉框的形式呈现给用户，列出他们输入内容的可能匹配。
```

File: files/zh-cn/learn/forms/other_form_controls/index.md [hunk 10]
```diff
@@ -0,0 +1,325 @@
...
+
+> **备注：** 你可以在 GitHub 上找到本节的例子，如 [other-examples.html](https://github.com/mdn/learning-area/blob/main/html/forms/native-form-widgets/other-examples.html)（[也可以看看在线演示](https://mdn.github.io/learning-area/html/forms/native-form-widgets/other-examples.html)）。
+
+### 计量器和进度条
+
+计量器和进度条是数值的视觉表现。
+
+#### 进度条
+
+一个进度条表示一个随时间变化的竖直，最高可以达到 {{htmlattrxref("max","progress")}} 属性所指定的最大值。这样的条是使用 {{ HTMLElement("progress")}} 元素创建出来的。
```
Comment (jasonren0403) [inline files/zh-cn/learn/forms/other_form_controls/index.md:250]: fix typo
```suggestion
进度条表示一个随时间变化的数值，最高可以达到 {{htmlattrxref("max","progress")}} 属性所指定的最大值。这样的条是使用 {{ HTMLElement("progress")}} 元素创建出来的。
```

File: files/zh-cn/learn/forms/other_form_controls/index.md [hunk 11]
```diff
@@ -0,0 +1,325 @@
...
+
+  - 如果 {{htmlattrxref("optimum","meter")}} 在下半部分范围内，下半部分范围被认为是首选部分，中间部分范围被认为是平均部分，而上半部分范围被认为是最差的部分。
+  - 如果 {{htmlattrxref("optimum","meter")}} 在中间部分范围内，下半部分范围被认为是平均部分，中间部分范围被认为是首选部分，而上半部分范围也被认为是平均部分。
+  - 如果 {{htmlattrxref("optimum","meter")}} 在上半部分范围内，下半部分范围被认为是最差的部分，中间部分范围被认为是平均部分，而上半部分范围被认为是首选部分。
+
+所有实现了 {{HTMLElement("meter")}} 元素的浏览器使用这些值来改变计量器的颜色：
+
+- 如果当前值位于首选范围，则计量器为绿色。
+- 如果当前值位于平均范围，则计量器为黄色。
+- 如果当前值位于最差范围，则计量器为红色。
```
Comment (jasonren0403) [inline files/zh-cn/learn/forms/other_form_controls/index.md:282]: 优化表达
```suggestion
- 如果当前值位于首选范围，则计量器显示为绿色。
- 如果当前值位于平均范围，则计量器显示为黄色。
- 如果当前值位于最差范围，则计量器显示为红色。
```

File: files/zh-cn/learn/forms/other_form_controls/index.md [hunk 12]
```diff
@@ -0,0 +1,325 @@
...
+<textarea cols="30" rows="8"></textarea>
+```
+
+这行代码的渲染效果如下所示：
+
+{{EmbedLiveSample("多行文本域", 120, 160)}}
+
+`<textarea>` 元素和普通的单行文本域的主要区别是，用户可以在要提交的数据中包含硬换行（如按下回车键产生）。
+
+`<textarea>` 也需要关闭标签，其中要包含的默认文字需要放在开闭标签之间。作为对比，{{HTMLElement("input")}} 是不包含关闭标签的空元素，它的默认值应该在 [`value`](/zh-CN/docs/Web/HTML/Attributes/value) 属性中指定。
```
Comment (yin1999) [inline files/zh-cn/learn/forms/other_form_controls/index.md:43]:
```suggestion
`<textarea>` 也需要关闭标签，其中要包含的默认文字需要放在开闭标签之间。作为对比，{{HTMLElement("input")}} 是不包含关闭标签的空元素，它的默认值应该在 [`value`](/zh-CN/docs/Web/HTML/Element/input#value) 属性中指定。
```

mdn/content#20017

**Review by yin1999 — APPROVED** [review]

**Pull request merged**